micromanager_gui/main_window.py

```diff
@@ -148,92 +156,105 @@ def add_frame_explorer(self, name, array):
-                #zpositions
-                if event_index_z > 0 and event_index_c == 0 and event_index_p == 0 and event_index_t == 0:
-                    empty_im = np.empty(((1,)*len(sequence.axis_order) + image.shape), dtype=np.uint16) 
-                    layer.data = np.concatenate((layer.data,empty_im), axis=-4)
+            for a, v in enumerate(im_idx):
+                self.viewer.dims.set_point(a, v)
+
+            #save each image in the temp folder, 
+            image_name = f'{im_idx}.tif'
+            savefile = temp_folder / image_name
+
+            tifffile.tifffile.imsave(str(savefile), image)
+
+            #and delete temp folder and files when mda is done
+            if tuple(i+1 for i in im_idx) == seq.shape:
+
+                shutil.rmtree(temp_folder.parent)
                 
-                #channels
-                if event_index_c > 0 and event_index_z == 0 and event_index_p == 0 and event_index_t == 0:
-                    empty_im = np.empty(((1,)*(len(sequence.axis_order)-2) + (z_stack_length,1,) + image.shape), dtype=np.uint16)
-                    layer.data = np.concatenate((layer.data,empty_im), axis=-3)
-
-                #xypositions
-                if event_index_p > 0 and event_index_c == 0 and event_index_z == 0 and event_index_t == 0:
-                    empty_im = np.empty(((1,)*(len(sequence.axis_order)-2) + (z_stack_length,c_stack_length,) + image.shape), dtype=np.uint16)
-                    layer.data = np.concatenate((layer.data,empty_im), axis=-5)
-
-                #timepoints
-                if event_index_t > 0 and event_index_z == 0 and event_index_c == 0 and event_index_p == 0:
-                    empty_im = np.empty(((1,p_stack_length,z_stack_length,c_stack_length,) + image.shape), dtype=np.uint16)
-                    layer.data = np.concatenate((layer.data,empty_im), axis=-6)
-
-            layer.data[event_index_t,event_index_p,event_index_z,event_index_c, ...] = image
-         
-            #set which dimension to display in napari viewer -> viewer.dims.set_point(axis, index)
-            self.viewer.dims.set_point(3, event_index_c)
-            self.viewer.dims.set_point(2, event_index_z)
-            self.viewer.dims.set_point(1, event_index_p)
-            self.viewer.dims.set_point(0, event_index_t)   
+            #save layer when acquisition is finished
+            if self.mda.save_groupBox.isChecked():
 
-        except KeyError:
+                if tuple(i+1 for i in im_idx) == seq.shape:
+
+                    name = self.viewer.layers.selection.active.name
+
+                    save_path = Path(self.mda.dir_lineEdit.text()) / name
+            
+                    self.viewer.layers[name].save(str(save_path))
+
+
+        except StopIteration:
+
+            if self.mda.save_groupBox.isChecked():
+                file_name = self.mda.fname_lineEdit.text()
+                layer_name = f"{file_name}_{datetime.now().strftime('%H:%M:%S')}"
```

Comment:
let's not include the entire filename in the layer name.  If anything, use `os.path.basename`

Reply:
if you have addressed these comments are confident with the fix... click the "resolve conversation" button at the bottom of the comment so I know I don't have check to see if it's still there.  If you feel like i might want to remember what the issue was, you can them them, but perhaps add a comment like "fixed here, etc..."   Otherwise, the reviewer needs to do more work to get themselves back in the mindset of what you're working on

Reply:
still there ...